cpp/include/cudf/io/orc_metadata.hpp

```diff
@@ -38,18 +51,130 @@ namespace io {
  * The following code snippet demonstrates how to read statistics of a dataset
  * from a file:
  * @code
- *  std::string filepath = "dataset.orc";
- *  auto result = cudf::read_orc_statistics(cudf::source_info(filepath));
+ *  auto result = cudf::read_raw_orc_statistics(cudf::source_info("dataset.orc"));
  * @endcode
  *
  * @param src_info Dataset source
  *
- * @return Decompressed ColumnStatistics blobs stored in a vector of vectors. The first element of
- * result vector, which is itself a vector, contains the name of each column. The second element
- * contains statistics of each column of the whole file. Remaining elements contain statistics of
- * each column for each stripe.
+ * @return ORC file statistics and column names.
+ */
+raw_orc_statistics read_raw_orc_statistics(source_info const &src_info);
+
+/**
+ * @brief Base class for different column statistics that include optional min and max values.
+ *
+ * Includes accessors for the min and max values.
+ */
+template <typename T>
+struct minmax_statistics {
+  std::unique_ptr<T> minimum;
+  std::unique_ptr<T> maximum;
+
+  auto has_minimum() const { return minimum != nullptr; }
+  auto has_maximum() const { return maximum != nullptr; }
+  auto *get_minimum() const { return minimum.get(); }
+  auto *get_maximum() const { return maximum.get(); }
```

Comment:
I would remove `get_` from all of the accessor methods. I like to follow the principle:
* Non-modifying functions (aka `const`) should be nouns
* Modifying functions should be verbs

Also, my preference is to omit the `*`:
```suggestion
  auto minimum() const { return minimum.get(); }
  auto maximum() const { return maximum.get(); }
```